src/main/java/com/google/devtools/build/lib/rules/proto/ProtoCompileActionBuilder.java

```diff
@@ -288,21 +280,16 @@ private static String getOutputDirectory(RuleContext ruleContext) {
       result.add("--disallow_services");
     }
     if (checkStrictImportPublic) {
-      NestedSet<Pair<Artifact, String>> protosInExports =
-          protoInfo.getExportedProtoSourcesImportPaths();
-      if (protosInExports.isEmpty()) {
+      NestedSet<ProtoSource> publicImportSources = protoInfo.getPublicImportSources();
+      if (publicImportSources.isEmpty()) {
         // This line is necessary to trigger the check.
         result.add("--allowed_public_imports=");
```

Comment:
A test case in our internal test battery fails. I can't export it at the moment, but the BUILD file it uses is as follows:

```
proto_library(
  name = 'myProto',
  srcs = ['myProto.proto'],
)
```

then verifies that its proto compile action has `--allowed_public_imports=`. This fails, and instead, it has `--allowed_public_imports test/myProto.proto`. I think I know why (see the other comment)